examples/js/DragNDrop.js

```diff
@@ -0,0 +1,173 @@
+                } else if (extension.type == _XML) {
+                    data = new window.DOMParser().parseFromString(data, 'text/xml');
+                }
+
+                extension.parser(data, {
+                    buildExtent: true,
+                    crsIn: 'EPSG:4326',
+                    crsOut: (extension.mode == _GEOMETRY ? _view.referenceCrs : _view.tileLayer.extent.crs),
+                    mergeFeatures: true,
+                    withNormal: (extension.mode == _GEOMETRY),
+                    withAltitude: (extension.mode == _GEOMETRY),
+                }).then(function _(result) {
+                    var dimensions = result.extent.dimensions();
+
+                    var source = new itowns.FileSource({
+                        parsedData: result,
+                        projection: 'EPSG:4326',
+                    });
+
+                    var randomColor = Math.round(Math.random() * 0xffffff);
+
+                    var layer;
+                    if (extension.mode == _COLOR) {
+                        layer = new itowns.ColorLayer(file.name, {
+                            transparent: true,
+                            style: {
+                                fill: {
+                                    color: '#' + randomColor.toString(16),
+                                    opacity: 0.7,
+                                },
+                                stroke: {
+                                    color: '#' + randomColor.toString(16),
+                                },
+                            },
+                            source: source,
+                        });
+                    } else if (extension.mode == _GEOMETRY) {
+                        layer = new itowns.GeometryLayer(file.name, new itowns.THREE.Group(), {
+                            update: itowns.FeatureProcessing.update,
+                            convert: itowns.Feature2Mesh.convert({
+                                color: new itowns.THREE.Color(randomColor),
+                                // Set the extrusion according to the size of
+                                // the extent containing the data; this quick
+                                // formula is totally arbitrary.
+                                extrude: dimensions.x * dimensions.y / 1e6,
+                            }),
+                            source: source,
+                            opacity: 0.7,
+                        });
+                    } else {
+                        throw new Error('Mode of file not supported, please add it using DragNDrop.register');
+                    }
+
+                    _view.addLayer(layer);
+
+                    // Move the camera to the first vertex
+                    itowns.CameraUtils.animateCameraToLookAtTarget(_view, _view.camera.camera3D, {
+                        coord: new itowns.Coordinates(result.crs, result.features[0].vertices),
+                        range: 25000,
```

Comment:
could youcompute range in function the dimension of extent?

Reply:
Done